mica/stats/acq_stats.py

```diff
@@ -0,0 +1,572 @@
+table_file = os.path.join(SKA, 'data', 'acq_stats', 'acq_stats.h5')
+
+
+def deltas_vs_obc_quat(vals, times, catalog):
+    # Ignore misalign
+    aca_misalign = np.array([[1, 0, 0], [0, 1, 0], [0, 0, 1]])
+    q_att = Quat(np.array([vals['AOATTQT1'],
+                           vals['AOATTQT2'],
+                           vals['AOATTQT3'],
+                           vals['AOATTQT4']]).transpose())
+    Ts = q_att.transform
+    acqs = catalog
+
+    R2A = 206264.81
+
+    dy = {}
+    dz = {}
+    star_info = {}
+    for slot in range(0, 8):
+        if slot not in acqs['slot']:
+            continue
+        agasc_id = acqs[acqs['slot'] == slot][0]['id']
+        if agasc_id is None:
+            logger.info("No agasc id for slot {}, skipping".format(slot))
+            continue
+        try:
+            star = agasc.get_star(agasc_id)
+        except:
+            logger.info("agasc error on slot {}:{}".format(
+                    slot, sys.exc_info()[0]))
+            continue
+        ra = star['RA']
+        dec = star['DEC']
+        if (star['PM_RA'] != -9999 or star['PM_DEC'] != -9999):
+            # Compute the multiplicative factor to convert from
+            # the AGASC proper motion field to degrees.  The AGASC PM
+            # is specified in milliarcsecs / year, so this is
+            # dyear * (degrees / milliarcsec)
+            dyear = ((DateTime(times[0]).secs
+                     - DateTime("{}:001".format(int(star['EPOCH']))).secs)
+                     / (86400 * 365.25))
+            pm_to_degrees = dyear / (3600. * 1000.)
+            if star['PM_RA'] != -9999:
+                ra_scale = np.cos(np.radians(dec))
+                ra = star['RA'] + star['PM_RA'] * pm_to_degrees / ra_scale
+            if star['PM_DEC'] != -9999:
+                dec = star['DEC'] + star['PM_DEC'] * pm_to_degrees
+        star_pos_eci = Ska.quatutil.radec2eci(ra, dec)
+        d_aca = np.dot(np.dot(aca_misalign, Ts.transpose(0, 2, 1)),
+                       star_pos_eci).transpose()
+        yag = np.arctan2(d_aca[:, 1], d_aca[:, 0]) * R2A
+        zag = np.arctan2(d_aca[:, 2], d_aca[:, 0]) * R2A
+        dy[slot] = vals['AOACYAN{}'.format(slot)] - yag
+        dz[slot] = vals['AOACZAN{}'.format(slot)] - zag
+        star_info[slot] = star
+
+    return dy, dz, star_info
+
+
+def get_delta_quat(eng_data, times, manvr):
+    guide = eng_data[times >= DateTime(manvr.guide_start).secs - 1]
+    first_guide = guide[0]
+    # skip the duplicate and go to index 2
+    second_guide = guide[2]
+    q1 = Quat([first_guide['AOATTQT1'],
+               first_guide['AOATTQT2'],
+               first_guide['AOATTQT3'],
+               first_guide['AOATTQT4']])
+    q2 = Quat([second_guide['AOATTQT1'],
+               second_guide['AOATTQT2'],
+               second_guide['AOATTQT3'],
+               second_guide['AOATTQT4']])
+    dq = q2 / q1
+    dot_q = np.sum(q1.q * q2.q)
+    if dot_q > 1:
+        dot_q = 1.0
+    return dq, dot_q
+
+
```

Comment:
I think the broadcasting wasn't correct for this application in mica.quaternion.  This should eventually get pushed back there if it is the *right thing*.